### DISCUSSION TITLE
Unwanted line swapping when helix is run within tmux

### DISCUSSION THREAD

**Discussion by chrisolsen** [helix-editor/helix#12062] [Q&A]
**This issue is one that I only experience when using Helix within Tmux.**

For some reason as I am typing I quite often get unwanted line swaps, which often goes unnoticed until either syntax highlights being to appear or I run a failing build. 

The other issue is I can't replicate what I do. It often seems to happen when I have a line or two selected, I click `esc`, then move my cursor to some location to insert some text. I can't even replicate the issue immediately after it happens, so it seems like the speed at which the keys are pressed is part of the process.

Attached is a video of the issue, that to reproduce, took a while of just clicking the same key combinations until it happened.

Does anyone know what is happening and have a fix?

https://github.com/user-attachments/assets/234c8093-5026-4a30-b8d0-3747f24ace1b

**Comment by TornaxO7** [comment]
Hm... which helix version are you using? Does it happen on the master branch as well?

**Reply by chrisolsen** [reply]
I am using 24.07 `079f5442` and haven't tried the master branch, but I can try.

**Reply by chrisolsen** [reply]
Just tried the lastest `6cca9826` and I still got the issue

**Reply by TornaxO7** [reply]
I haven't experienced anything like this sadly :(
